src/Symfony/Component/Security/Core/Authorization/AccessDecisionManager.php

```diff
@@ -68,142 +95,44 @@ public function decide(TokenInterface $token, array $attributes, $object = null/
             throw new InvalidArgumentException(sprintf('Passing more than one Security attribute to "%s()" is not supported.', __METHOD__));
         }
 
-        return $this->{$this->strategy}($token, $attributes, $object);
+        return $this->strategy->decide(
+            $this->collectResults($token, $attributes, $object)
+        );
     }
 
     /**
-     * Grants access if any voter returns an affirmative response.
+     * @param mixed $object
      *
-     * If all voters abstained from voting, the decision will be based on the
-     * allowIfAllAbstainDecisions property value (defaults to false).
+     * @return \Traversable<int>
      */
-    private function decideAffirmative(TokenInterface $token, array $attributes, $object = null): bool
+    private function collectResults(TokenInterface $token, array $attributes, $object): \Traversable
     {
-        $deny = 0;
         foreach ($this->voters as $voter) {
             $result = $voter->vote($token, $object, $attributes);
-
-            if (VoterInterface::ACCESS_GRANTED === $result) {
-                return true;
-            }
-
-            if (VoterInterface::ACCESS_DENIED === $result) {
-                ++$deny;
-            } elseif (VoterInterface::ACCESS_ABSTAIN !== $result) {
+            if (!(self::VALID_VOTES[$result] ?? false)) {
```

Comment:
this won't work to report a deprecation when a voter returns a boolean by mistake, as it assumes that the returned result can be used as an array key.

Reply:
Fixed and covered by a test.